### PR TITLE
feat(dropdown) abstract positioning variants, add trigger for both click and hover

### PR DESCRIPTION
**Pull request by brandonlilly**
**Before submitting a pull request,** please make sure the following is done:

I have done **all** of the following:

- [x] Added a top level class to all my components `'.anchor-[COMPONENT NAME]'`.
- [x] Used [conventional commits](https://www.conventionalcommits.org) for all work.
- [ ] Tested my solution on Mobile & Tablet.
- [x] Wrote [unit tests](https://jestjs.io/docs/en/getting-started) for states and all behavior (`npm test`) and passed coverage thresholds.
- [x] Updated snapshots for all permutations (`npm test -- -u`).
- [x] Accounted for hover, focus, blur, visited, & error states because they are not edge cases.
- [x] Created TODOs for known edge cases.
- [ ] Documented all of my changes (inline & doc site).
- [x] Made sure that all accessibility errors are resolved.
- [x] Added [stories](https://storybook.js.org/docs/basics/introduction/) with knobs for all possible configurations.
- [x] De-linted and ran [prettier](https://github.com/prettier/prettier) (`npm run pretty`) on my code.
- [x] Added name to OWNERS file for all new components
- [x] If adding a new component, add its export to the rollup config
- [x] package.json version is bumped (if necessary)

---------
**Outline your feature or bug-fix below**

My code does...

### CODE REVIEW

File: src/Tooltip/Tooltip.component.tsx [hunk 1]
```diff
@@ -4,20 +4,9 @@ import * as React from 'react';
...
-    | 'rightEnd'
-    | 'bottomEnd'
-    | 'bottom'
-    | 'bottomStart'
-    | 'leftEnd'
-    | 'left'
-    | 'leftStart';
+// ANCHOR
+import { positionVariants, Position } from '../utils/position/position';
+export { Position } from '../utils/position/position';
```
Comment (zero0Halo) [inline src/Tooltip/Tooltip.component.tsx:9]: Any reason why `Position` is being exported here? I noticed it was also being exported from `DropDown` too.

File: src/DropDown/DropDown.stories.tsx [hunk 2]
```diff
@@ -127,20 +129,49 @@ storiesOf('Components/DropDown', module)
...
+                                        text('arrowIndent', '') || undefined
+                                    }
+                                    arrowSize={
+                                        text('arrowSize', '') || undefined
+                                    }
+                                    spacing={text('spacing', '') || undefined}
                                 >
-                                    <a>{text('Dropdown Label', 'Dropdown')}</a>
+                                    <Button>
+                                        {text('Dropdown Text', 'Dropdown')}
```
Comment (zero0Halo) [inline src/DropDown/DropDown.stories.tsx]: I would actually not have this text configurable. It makes it seem as if it's a prop for `DropDown` but it isn't. I questioned why `DropDown` would have configurable text based on what I knew about how it was put together, I thought maybe you changed something. If it confused me, I think it'll confuse the average user too.

File: src/utils/position/position.ts [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+            });
+        case 'topEnd':
+        default:
+            return css({
+                right: 0,
+                top: 0,
+                transform: `translate3d(0, -${overlayHeight + spacing}px, 0)`,
+            });
+    }
+};
```
Comment (zero0Halo) [inline src/utils/position/position.ts:105]: Love that you moved this out of `Tooltip` as its own resource to draw from.

File: src/DropDown/DropDown.stories.tsx (hunk 2 above)
Comment (brandonlilly) [reply src/DropDown/DropDown.stories.tsx]: I agree, but I also want a knob for the text so that I can easily test the behavior for various sizes. Maybe I could relabel it for clarity?

File: src/Tooltip/Tooltip.component.tsx (hunk 1 above)
Comment (brandonlilly) [reply src/Tooltip/Tooltip.component.tsx:9]: It was the fastest way to extract the position utility without having to redo imports from this file.

But ultimately I figured it was alright to leave it here and be able to import from this file to see what positions exist. I don't feel strongly about it however, just seems more convenient.

File: src/DropDown/DropDown.stories.tsx (hunk 2 above)
Comment (zero0Halo) [reply src/DropDown/DropDown.stories.tsx]: Ah, makes sense from a testing perspective. I think any way you can denote that (even a comment in the story) would be good.

File: src/Tooltip/Tooltip.component.tsx (hunk 1 above)
Comment (zero0Halo) [reply src/Tooltip/Tooltip.component.tsx:9]: Since you're doing it more for convenience in the developing of the stories I'm not going to be picky about it. My concern was more about the ultimate export of two components, if `Position` was being exported twice that could be a problem. As that's not what's going on, I'm not worried.

**Review by zero0Halo — APPROVED** [review]
LGTM
